web/services/vms/images/devbox/Dockerfile

```
@@ -122,14 +131,65 @@ ENV AGENT_BROWSER_EXECUTABLE_PATH=/usr/bin/google-chrome-stable
+# never a second copy. GL/Vulkan/xkb libraries are what Ghostty and Chrome
+# render with under Xvnc. Ghostty ships no upstream .deb; the community build
+# for Ubuntu 24.04 is pinned by release tag.
+ARG CMUX_IMAGE_DESKTOP_PACKAGES="tigervnc-standalone-server tigervnc-tools openbox tint2 thunar feh novnc websockify at-spi2-core dbus-x11 libglib2.0-bin x11-xserver-utils xdg-utils adwaita-icon-theme libgl1-mesa-dri mesa-vulkan-drivers libvulkan1 libxkbcommon0 libxkbcommon-x11-0 netcat-openbsd"
+ARG CMUX_IMAGE_GHOSTTY_DEB_URL=https://github.com/mkasberg/ghostty-ubuntu/releases/download/1.3.1-0-ppa2/ghostty_1.3.1-0.ppa2_amd64_24.04.deb
+RUN apt-get update && apt-get install -y --no-install-recommends $CMUX_IMAGE_DESKTOP_PACKAGES \
+  && rm -rf /var/lib/apt/lists/* \
+  && { [ -e /usr/share/novnc/index.html ] || ln -s vnc.html /usr/share/novnc/index.html; } \
+  && command -v Xtigervnc && command -v vncconfig && command -v websockify && command -v openbox && command -v tint2 \
+  && command -v dbus-launch && command -v dbus-send && command -v gdbus && command -v feh && command -v thunar \
+  && { test -x /usr/libexec/at-spi-bus-launcher || test -x /usr/lib/at-spi2-core/at-spi-bus-launcher; }
+RUN curl -fsSL -o /tmp/ghostty.deb "$CMUX_IMAGE_GHOSTTY_DEB_URL" \
```

Hadolint (2.15.1)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)